File: app/src/main/java/com/ikem/nwodo/cryptonyte/main/MainActivityModule.java
package com.ikem.nwodo.cryptonyte.main;

import android.content.Context;

import com.ikem.nwodo.cryptonyte.CryptoAdapter;
import com.ikem.nwodo.cryptonyte.di.scope.MainActivityScope;

import dagger.Module;
import dagger.Provides;

//@Module
public class MainActivityModule {

    private final MainActivity mainActivity;

    public MainActivityModule(MainActivity mainActivity){
        this.mainActivity = mainActivity;
    }

}

File: app/src/main/java/com/ikem/nwodo/cryptonyte/di/MainActivityModule.kt
package com.ikem.nwodo.cryptonyte.di;

import com.ikem.nwodo.cryptonyte.MainActivity
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract class MainActivityModule {
    @ContributesAndroidInjector(
            modules = [FragmentBuildersModule::class]
    )
    abstract fun contributeMainActivity() : MainActivity
}
